src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.DiagnosticProvider.cs

```diff
@@ -31,6 +35,30 @@ public abstract class DiagnosticProvider
             /// This includes both document-level diagnostics for all documents in the given project and project-level diagnostics, i.e. diagnostics with no source location, in the given project. 
             /// </summary>
             public abstract Task<IEnumerable<Diagnostic>> GetAllDiagnosticsAsync(Project project, CancellationToken cancellationToken);
+
+            internal virtual Task<ImmutableDictionary<Document, ImmutableArray<Diagnostic>>> GetDocumentDiagnosticsToFixAsync(
```

Comment:
This API seems weird - the FixAllContext diagnostic provider calls back into the batch fixer to provide diagnostics. We should instead just throw NotImplementedException here and remove the virtual methods GetDocumentDiagnosticsToFixAsync and GetProjectDiagnosticsToFixAsync being added here.